_includes/markdown/Typescript.md

```diff
@@ -0,0 +1,222 @@
+3. `strictNullChecks: true` - [Definition](https://www.typescriptlang.org/tsconfig/#strictNullChecks) - forces handling both `undefined` and `null`
+4. `noUnusedParameters: true` - [Definition](https://www.typescriptlang.org/tsconfig/#noUnusedParameters) - does not allow unused parameters to remain in code
+
+We also have a recommended [tsconfig.json](https://github.com/10up/headstartwp/blob/develop/projects/wp-nextjs/tsconfig.json) for Next.js projects.
+
+## TypeScript and React
+
+### Use `type` for React Props
+
+While `interface` and `type` are mostly interchangeable, prefer `type` over `interfaces` unless you expect the props to be the base for other components' props.
+
+```tsx
+type BlocksProps = {
+	html: string;
+};
+
+export const Blocks: React.FC<BlockProps> = ({ html }) => {
+    // render blocks
+}
+```
+
+### Use `React.FC` to type React Components
+
+When typing React components prefer using `React.FC`. While in the past it was problematic because of implicitly accepting `children` as an optional prop, since React 18 it [no longer does that](https://www.totaltypescript.com/you-can-stop-hating-react-fc). Using React.FC ensures your component is correctly typed, including its return type.
+
+```tsx
+import * as React from 'react';
+
+type MyComponentProps = {
+	/* */
+};
+
+export const MyComponent: React.FC<MyComponentProps> = (props) => {
+	// component logic
+}
+```
+**Note**: If using React < 17, be aware that `children` will be explicitly set for all components using `React.FC`. This can cause some TS errors when upgrading React to 18+, if a component is passed `children` but does not explicitly declare it as a prop.
+
+The [React TypeScript Cheatsheet](https://react-typescript-cheatsheet.netlify.app/) is a good resource to check how to type specific things in React (e.g. Forms, Events etc).
+
+### Use function default arguments instead of `defaultProps`
+
+`defaultProps` has been deprecated in React 19, therefore prefer using function default arguments.
+
+```tsx
+type MyComponentProps = {
+    title?: string;
+};
+
+export const MyComponent = ({ title = 'Default Title' }: MyComponentsProps) => {
+    // ...
+}
+```
+
+Make sure the ESLint rule `react/require-default-props` is set up to look for `defaultArguments` e.g `'react/require-default-props': ['error', { functions: 'defaultArguments' }]`;
+
+### Prefer explicitly declaring `children` instead of using `PropsWithChildren`
+
+`PropsWithChildren` will make `children` optional, but sometimes you do want to be very explicit and make it required. Therefore we generally recommend declaring `children` explicitly.
+
+```tsx
+type LayoutProps = {
+    children: React.Node;
+};
+
+// If Layout is not passed a children, TS will catch the error
+const Layout = (props: LayoutProps) => {
```

Comment:
@nicholasio should this be rewritten using React.FC as recommended above?

Reply:
Made a quick follow-up commit to update this thanks